peer-monitoring-service/client/src/peer_states/latency_info.rs

```diff
@@ -61,13 +76,23 @@ impl LatencyInfoState {
         // Update the number of ping failures for the request tracker
         self.request_tracker.write().record_response_failure();
 
-        // TODO: If the number of ping failures is too high, disconnect from the node
         let num_consecutive_failures = self.request_tracker.read().get_num_consecutive_failures();
         if num_consecutive_failures >= self.latency_monitoring_config.max_latency_ping_failures {
+            let reason = "Too many ping failures occurred for the peer!";
             warn!(LogSchema::new(LogEntry::LatencyPing)
                 .event(LogEvent::TooManyPingFailures)
                 .peer(peer_network_id)
-                .message("Too many ping failures occurred for the peer!"));
+                .message(reason));
+
+            // If the number of ping failures is too high, disconnect from the node
+            let service_client = self.peer_monitoring_service_client.clone();
+            let peer_network_id = *peer_network_id.clone();
+            tokio::spawn(async move {
+                // TODO revisit if tokio spawn is OK here :(, otherwise very painful
```

Comment:
Yeah, it's probably fine to do this. We'll just need to test it, i.e., invoke random disconnects in forge to verify that the connection is cut.